### ISSUE TITLE
Pay button does not have loading state after clicking pay

### ISSUE THREAD

**Issue by jcpof** [Adyen/adyen-web#1488]
**Describe the bug**

**To Reproduce**
1. create a drop in component
2.  fill out form
3.  click pay button

**Expected behavior**
pay button should be disabled and show a loading spinner

**Desktop (please complete the following information):**
chrome, safari, edge

**Smartphone (please complete the following information):**
mac, windows, iphone

**Additional context**
this appears to happen sometime †after v4.5.0 .

**Comment by jcpof** [comment]
i have tried 5.0.0 and 5.6.1. , 5.7.0 and all have this issue @ribeiroguilherme . i can click the pay button multiple times resulting in multiple transactions

**Comment by ribeiroguilherme** [comment]
Can you show how you are creating the drop-in and its event handlers that you attached?

**Comment by carter-roller** [comment]
I'm experiencing a similar issue, but only when `setStatusAutomatically` is false.  We already have a working implementation of V4, but with V5 I can't figure out how to set the loading state of the button.  We don't want to use `setStatusAutomatically` because we want to handle refusals differently.  I tried using `onSubmit` and `setStatus('loading')` which did work, however, that seemed to stop/break the payment flow.  I've also tried true for `setStatusAutomatically` and then `onPaymentCompleted` doing `setStatus('ready')`, however, that loses the state of the card information entered.

With V5, how do we achieve the loading/disabled button and retain card state on refusals?

**Comment by ribeiroguilherme** [comment]
Hi @carter-roller ,

With sessions, you should not provide the `onSubmit` callback - that will break the flow indeed. You should use different callbacks for this case:

````
 const checkout = await AdyenCheckout({
        setStatusAutomatically: false,
        ...
        // Events
        beforeSubmit: (data, component, actions) => {
            component.setStatus('loading');
            actions.resolve(data);
        },
        onPaymentCompleted: (result, component) => {
            if (result.resultCode === 'Refused') {
                component.setStatus('ready');
                // handle failure
            }
            // handle success
        },
```

There was an issue with how drop-in was handling the `status`  which is fixed on #1394 , and released on v5.6.2. I recommend you to use the latest version of the library. I tested it and it is keeping the card state on refusal.

**Comment by ribeiroguilherme** [comment]
I am closing this issue for now. Let us know if the problem isn't fixed. I can always re-open the ticket.
